db/pending_migration_cleanups/20230516160033_delete_worldwide_organisation_about_pages.rb

```diff
@@ -0,0 +1,27 @@
+class DeleteWorldwideOrganisationAboutPages < ActiveRecord::Migration[7.0]
+  class Edition < ApplicationRecord
+  end
+
+  class CorporateInformationPage < Edition
+    ABOUT_US_TYPE_ID = 20
+  end
+
+  class WorldwideOrganisation < ApplicationRecord
+    has_many :edition_worldwide_organisations
+    has_many :corporate_information_pages, through: :edition_worldwide_organisations, source: :edition, class_name: "::CorporateInformationPage"
+
+    def about_us_page_editions
+      corporate_information_pages.where(corporate_information_page_type_id: CorporateInformationPage::ABOUT_US_TYPE_ID)
+    end
+  end
+
+  def up
+    WorldwideOrganisation.all.each do |worldwide_organisation|
+      worldwide_organisation.about_us_page_editions.destroy_all
```

Comment:
Will this delete attachments on old editions, too?

I've only glanced, but it looks like it might be necessary to call `delete_all_attachments`:

https://github.com/alphagov/whitehall/blob/ab98fb14d01fddd38abf79a430fb68dbcf365de8/app/models/attachable.rb#L178-L180

because the `has_many` association doesn't have `dependent: :destroy`

Reply:
Hmm. Good question. Since we've moved (rather than copied) the attachments from each of the about page published edition to the relevant WO, I don't think we want to delete those attachments. However, we *do* want to delete the attachments associated with about page non-published editions which I had overlooked. Perhaps we need two steps, one where we delete the about page published editions (but not their attachments) and one where we delete the about page non-published editions *and* their attachments...?

@chrisroos Are you happy to look into this? Let me know if you want to pair on it.

Reply:
> two steps, one where we delete the about page published editions (but not their attachments) and one where we delete the about page non-published editions and their attachments

Won't the 'current' published Edition technically have _zero_ attachments? Since you've already moved them (rather than copied). So calling `delete_all_attachments` should technically do nothing, since they're no longer associated to that Edition? 🤔 Or maybe I'm missing something.

Reply:
Ah, yes - you're right - thanks!

Reply:
Hi @ollietreend and @floehopper - I believe I've addressed this in https://github.com/alphagov/whitehall/pull/7674/commits/c216f8f1945c923052bd7185c5e05b26121b520f. I'm fairly confident it's working as expected but would really appreciate a second or third pair of eyes on it given the destructive nature of the migration.

It takes about 15 mins to run on my machine. Is that an acceptable amount of time for a standard Rails migration to run? Or should we make this a data migration when we're happy to run it?